runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerCommand.java

```diff
@@ -147,17 +192,26 @@ private String runShortCommand(List<String> invocation)
               BufferedReader reader =
                   new BufferedReader(
                       new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
-              return reader.lines().collect(Collectors.joining());
-            });
-    // NOTE: We only consume the error string in the case of an error.
-    CompletableFuture<String> errorFuture =
-        CompletableFuture.supplyAsync(
-            () -> {
-              BufferedReader reader =
-                  new BufferedReader(
-                      new InputStreamReader(process.getErrorStream(), StandardCharsets.UTF_8));
-              return reader.lines().collect(Collectors.joining());
+              return reader.lines().collect(Collectors.joining(delimiter));
             });
+    CompletableFuture<String> errorFuture;
```

Comment:
Wouldn't a better place for this and all following lines to be in the 

```java
if (exitCode != 0) {
```

block? The `errorFuture` and `errorString` are not used elsewhere.